### PR TITLE
III-5749 Add labels data to behat tests and fix failing test

### PR DESCRIPTION
**Pull request by JonasVHG**
### Added

- Added Labels Data setup on `LabelSteps`
- Added `getLabel()` on `LabelSteps`
- Add step  `When I patch the label with :id and :command` on `LabelSteps`

### Changed

- Make sure LabelData is created when running tests for `labels/get`

Related PR: https://github.com/cultuurnet/udb3-acceptance-tests/pull/193

---
Ticket: https://jira.uitdatabank.be/browse/III-5749

### CODE REVIEW

File: features/Steps/LabelSteps.php [hunk 1]
```diff
@@ -34,6 +34,58 @@ public function iCreateAnInvisibleLabelWithARandomNameOfCharacters(int $nrOfChar
         );
     }
 
+    /**
+     * @Given labels test data is available
+     */
+    public function labelsTestDataIsAvailable(): void
+    {
+        // Create "public-visible" if it doesn't exist yet and (re)set the right privacy and visibility in case its needed
+        $this->createLabel('public-visible', true, true);
```
Comment (LucWollants) [inline features/Steps/LabelSteps.php:43]: I would first check if the label exists and only create it when it does not exist. When it already exists there is no need to create it again, only making sure to set the correct privacy and visibility.

File: features/label/get.feature [hunk 2]
```diff
@@ -4,6 +4,7 @@ Feature: Test the UDB3 labels API
     Given I am using the UDB3 base URL
     And I am using an UiTID v1 API key of consumer "uitdatabank"
     And I am authorized as JWT provider v1 user "centraal_beheerder"
+    And labels test data is available
```
Comment (LucWollants) [inline features/label/get.feature:7]: I would create the labels inside the `beforeSuite` method here https://github.com/cultuurnet/udb3-backend/blob/master/features/Bootstrap/FeatureContext.php#L72

**Review by LucWollants — CHANGES_REQUESTED** [review]
Already some nice progress! I would only tweak it to become more robust and performant.

File: features/Steps/LabelSteps.php [hunk 3]
```diff
@@ -40,48 +40,66 @@ public function iCreateAnInvisibleLabelWithARandomNameOfCharacters(int $nrOfChar
     public function labelsTestDataIsAvailable(): void
     {
         // Create "public-visible" if it doesn't exist yet and (re)set the right privacy and visibility in case its needed
-        $this->createLabel('public-visible', true, true);
         $this->getLabel('public-visible');
+        if ($this->responseState->getStatusCode() === 404) {
+            $this->createLabel('public-visible', true, true);
```
Comment (LucWollants) [inline features/Steps/LabelSteps.php:45]: If the label needs to be created there is no need to patch the privacy and visibility.

Comment (JonasVHG) [reply features/Steps/LabelSteps.php:45]: Nice find.

File: features/Steps/LabelSteps.php [hunk 4]
```diff
@@ -50,4 +121,27 @@ private function createLabel(string $name, bool $visible, bool $public): void
...
+    {
+        $response = $this->getHttpClient()->patchJSON(
+            '/labels/' . $uuid,
+            $this->variableState->replaceVariables(
+                Json::encode([
+                    'command' => $command,
+                ])
+            )
+        );
+        $this->responseState->setResponse($response);
```
[the remark this answers is not in the document]
Comment (LucWollants) [reply features/Steps/LabelSteps.php:145]: Maybe even better is to change the scenarios that do the patch on a label with this reusable method. Something like `And I patch the label <labelId> with command <command>`

[the remark this answers is not in the document]
Comment (JonasVHG) [reply features/Steps/LabelSteps.php]: Done, while I am at, I might be useful to also make a step `I get the label <labelId>`?

**Review by LucWollants — APPROVED** [review]

**Pull request merged**